plugin-management-cli/src/main/java/io/jenkins/tools/pluginmanager/cli/ConfigSetup.java

```diff
@@ -0,0 +1,162 @@
+        }
+
+        if (Files.exists(cfg.getPluginTxt().toPath())) {
+            try {
+                Scanner scanner = new Scanner(cfg.getPluginTxt(), StandardCharsets.UTF_8.name());
+                System.out.println("Reading in plugins from " + cfg.getPluginTxt().toString() + "\n");
+                while (scanner.hasNextLine()) {
+                    Plugin plugin = parsePluginLine(scanner.nextLine());
+                    plugins.add(plugin);
+                }
+            } catch (FileNotFoundException e) {
+                System.out.println("Unable to open " + cfg.getPluginTxt());
+            }
+        } else {
+            System.out.println(cfg.getPluginTxt() + " file does not exist");
+        }
+
+        cfg.setPlugins(plugins);
+    }
+
+    public Plugin parsePluginLine(String pluginLine) {
+        String[] pluginInfo = pluginLine.split(":");
+        String pluginName = pluginInfo[0];
+        String pluginVersion = "latest";
+        String pluginUrl = null;
+        if (pluginInfo.length >= 2) {
+            pluginVersion = pluginInfo[1];
+        }
+        if (pluginInfo.length == 3) {
+            pluginUrl = pluginInfo[2];
+        }
+
+        return new Plugin(pluginName, pluginVersion, pluginUrl);
+    }
+
+    public void getPluginDir() {
+        if (options.getPluginDir() == null) {
+            System.out.println("No directory to download plugins entered. " +
+                    "Will use default of " + Settings.DEFAULT_PLUGIN_DIR);
+            cfg.setPluginDir(Settings.DEFAULT_PLUGIN_DIR);
+        } else {
+            System.out.println("Plugin download location: " + options.getPluginDir());
+            cfg.setPluginDir(options.getPluginDir());
+        }
+    }
+
+
+    public void getWar() {
+        if (options.getJenkinsWar() == null) {
+            System.out.println("No war entered. Will use default of " + Settings.DEFAULT_JENKINS_WAR);
+            cfg.setJenkinsWar(Settings.DEFAULT_JENKINS_WAR);
+        } else {
+            System.out.println("Will use war file: " + options.getJenkinsWar());
+            cfg.setJenkinsWar(options.getJenkinsWar());
+        }
+    }
+
+
+    public void getUpdateCenters() {
+        String jenkinsUc = "";
+        if (!StringUtils.isEmpty(options.getJenkinsUc())) {
+            jenkinsUc = options.getJenkinsUc();
+            System.out.println("Using update center " + jenkinsUc + " specified with CLI option");
+        } else if (!StringUtils.isEmpty(System.getenv("JENKINS_UC"))) {
+            jenkinsUc = System.getenv("JENKINS_UC");
+            System.out.println("Using update center " + jenkinsUc + " from JENKINS_UC environment variable");
+        } else {
+            jenkinsUc = "https://updates.jenkins.io";
+            System.out.println(
+                    "No CLI option or environment variable set for update center, using default of " + jenkinsUc);
+        }
+
+        cfg.setJenkinsUc(jenkinsUc);
+
+        String jenkinsUcExperimental = "";
```

Comment:
shouldn't need an empty string default as you're always setting it, but I would suggest extracting the block below to a separate method.
```suggestion
        String jenkinsUcExperimental = getExperimentalUpdateCenter();
```
or
```suggestion
        String jenkinsUcExperimental = detectExperimentalUpdateCenter();
```